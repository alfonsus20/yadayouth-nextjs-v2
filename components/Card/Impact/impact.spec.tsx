import { render, screen } from "@testing-library/react";
import { ImpactCard } from "..";
import MockedImage from "../../../public/bg/home.jpg";

const mockedProps = {
  image: MockedImage,
  title: "Title",
  description: "Description",
  color: "yellow",
};

const className =
  "col-span-12 sm:col-span-6 md:col-span-3 md:min-h-[350px] flex flex-col p-8 md:pt-32 rounded-3xl relative shadow-lg";

jest.mock("next/image", () => ({
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img {...props} alt="image" />;
  },
}));

describe("Impact card unit test", () => {
  it("should render correctly", () => {
    render(<ImpactCard {...mockedProps} />);
    expect(screen.getByTestId("card-image")).toBeVisible();
    expect(screen.getByTestId("card-title")).toBeVisible();
    expect(screen.getByTestId("card-description")).toBeVisible();

    expect(screen.getByTestId("card-title")).toHaveTextContent("Title");
    expect(screen.getByTestId("card-description")).toHaveTextContent(
      "Description"
    );
  });

  it("should match snapshot", () => {
    const container = render(<ImpactCard {...mockedProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should render correct color", () => {
    const containerYellow = render(
      <ImpactCard {...mockedProps} color="yellow" />
    );
    expect(screen.getByTestId("card")).toHaveClass(
      `${className} bg-neon--yellow`
    );
    containerYellow.unmount();

    const containerBlue = render(<ImpactCard {...mockedProps} color="blue" />);
    expect(screen.getByTestId("card")).toHaveClass(
      `${className} bg-neon--blue`
    );
    containerBlue.unmount();

    const containerGreen = render(
      <ImpactCard {...mockedProps} color="green" />
    );
    expect(screen.getByTestId("card")).toHaveClass(
      `${className} bg-neon--green`
    );
    containerGreen.unmount();

    const containerOrange = render(
      <ImpactCard {...mockedProps} color="orange" />
    );
    expect(screen.getByTestId("card")).toHaveClass(
      `${className} bg-neon--orange`
    );
    containerOrange.unmount();
  });
});
