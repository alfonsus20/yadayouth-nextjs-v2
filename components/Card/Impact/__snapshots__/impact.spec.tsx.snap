// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Impact card unit test should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-span-12 sm:col-span-6 md:col-span-3 md:min-h-[350px] flex flex-col p-8 md:pt-32 rounded-3xl relative shadow-lg bg-neon--yellow"
        data-testid="card"
      >
        <div
          class="md:absolute md:-top-24 md:-right-4 mx-auto md:mx-0 md:w-56 md:h-56"
        >
          <img
            alt="image"
            data-testid="card-image"
            height="220"
            quality="100"
            src="home.jpg"
            width="220"
          />
        </div>
        <div
          class="text-white"
        >
          <h3
            class="font-semibold text-xl mb-1"
            data-testid="card-title"
          >
            Title
          </h3>
          <p
            class="text-lg"
            data-testid="card-description"
          >
            Description
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-span-12 sm:col-span-6 md:col-span-3 md:min-h-[350px] flex flex-col p-8 md:pt-32 rounded-3xl relative shadow-lg bg-neon--yellow"
      data-testid="card"
    >
      <div
        class="md:absolute md:-top-24 md:-right-4 mx-auto md:mx-0 md:w-56 md:h-56"
      >
        <img
          alt="image"
          data-testid="card-image"
          height="220"
          quality="100"
          src="home.jpg"
          width="220"
        />
      </div>
      <div
        class="text-white"
      >
        <h3
          class="font-semibold text-xl mb-1"
          data-testid="card-title"
        >
          Title
        </h3>
        <p
          class="text-lg"
          data-testid="card-description"
        >
          Description
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
